service/src/main.rs

```diff
@@ -390,7 +389,8 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 	}
 
 	// ------------------------------------------------------------------------
-	// Perform a remote attestation and get an unchecked extrinsic back.
+	// Init parentchain specific stuff. Needed for parentchain communication.
+	let last_synced_header = init_light_client(&node_api, enclave.clone()).unwrap();
```

Comment:
In my opinion, this is not the way to go. On the initialization level, we should separate the light client from the extrinsics factory. The extrinsics factory can be initialized at the very beginning (in the general `init` function), passing in the genesis hash from the node api.

That way, `init_light_client` really is only initialization the light client. And the extrinsics factory is available independently (e.g. for the teeracle, we then don't even call `init_light_client`, because we don't need it).

Reply:
That's a separate task: #1049